common/src/main/java/com/wynntils/screens/itemfilter/ItemFilterScreen.java

```diff
@@ -670,6 +670,10 @@ public void setFiltersForProvider(ItemStatProvider<?> provider, List<StatProvide
         }
 
         updateQueryString();
+
+        if (provider == selectedProvider) {
+            valueWidget.onFiltersChanged(filterPairs);
+        }
```

Comment:
I can't seem to understand why this is needed. It's also the reason as to why `BooleanValueWidget` needs the null check for filters.

Reply:
It's so when you interact with the provider button to add the any filter or remove all filters it updates the widget

Reply:
This has been removed now